src/sonic-yang-models/yang-models/sonic-device_metadata.yang

```diff
@@ -127,6 +127,11 @@ module sonic-device_metadata {
                     type string;
                 }
 
+                leaf sub_role {
+                    type string;
+                    description "sub_role indicates if ASIC is FrondEnd or BackEnd.";
```

Comment:
Does the field take FrontEnd or BackEnd alone? If yes should we restrict it with pattern?

Reply:
I have only seen one example with 'FrondEnd'. I also checked design doc. Seems 'Backend' and 'Frontend' are valid. https://github.com/Azure/SONiC/blob/master/doc/multi_asic/SONiC_multi_asic_hld.md#2432-device_metadata

I am not sure if I should restrict it with the pattern 'Front[Ee]nd|Back[Ee]nd' or simply a string instead.